core/src/main/java/org/apache/iceberg/actions/RewriteFileGroup.java

```diff
@@ -77,4 +78,9 @@ public String toString() {
         .add("numAddedFiles", addedFiles == null ? "Rewrite Incomplete" : Integer.toString(addedFiles.size()))
         .toString();
   }
+
+  @Override
+  public int compareTo(@Nonnull final RewriteFileGroup o) {
```

Comment:
this doesn't seem like it is required and i'm a bit dubious about having all rewriteFileGroups treated as equal. The Comparator approach below seems like the right thing to do

Reply:
This was mainly for the case when the user doesn't pass any option and we end up following the natural partition order. I have handled this in `rewriteGroupComparator()` now.